NewForce-Cohort-4/tabloidmvc-mean-beetle
# Edit a Post-ticket  #8

Description
The functionality to edit a post has been added.

Given the user is authenticated, when they click on "MY POSTS" in the navigation bar, they are taken to a page where they can view all of their posts.  They can then select the option to edit a Post. Then the user should be directed to a form and given the ability to change the Post's information
Given the user is finished updating the Post information
When they click the `Save` button
 Then the updated Post should be saved to the database
And the user should be redirected to the new Post's details page
Given the user has decided not to edit the Post
When they click the `Cancel` button
Then the user should be redirected back to the list page
Editable Information
 * Title
* Content
* Category
* Header Image URL (optional)
* Publication Date (optional)

Type of change:
 -New feature (non-breaking change which adds functionality)

Testing Instructions:
1.After making sure you have our repo cloned down into a local repository and have the database added, git fetch --all in the master branch to retrieve this branch named pp-posts that I have pushed up.

2.git checkout pp-posts to switch to my branch and open up Visual Studio

3.Run Update-Database in your Package Manager Console.

4.Then, run the application in a browser window. You should see the 'Home' page with a navigation bar at the top of the page. 'Log in' with the following credentials.
Username: admin@example.com


5.Once you're logged in, click on "MY POSTS" in the nav bar.

6.You will be directed to a page where you can view all of the posts that have been created by this user that is logged in(admin@example.com). You should see the edit button on the right hand side and be able to select it.

7.Once you click edit, a form should appear with all the previous information filled in from the post you selected. You can changed/edit any of the fields. Then click save, and the post will be saved to the database.

8.You will be redirected back to your list of posts. 

9. If you decide not to edit the post, hit cancel and you should be redirected back to your list of posts. 
The end.
